demisto_sdk/commands/validate/validate_manager.py

```diff
@@ -845,11 +859,14 @@ def get_modified_and_added_files(self, compare_type, prev_ver):
                     self.handle_error(error_message, error_code, file_path="General-Error", warning=True,
                                       drop_line=True)
 
-                if not self.no_configuration_prints:
+                if not self.no_configuration_prints and not self.staged:
                     click.echo("Collecting all local changed files against the content master")
-
-                # only changes against prev_ver (without local changes)
-                all_changed_files_string = run_command('git diff --name-status {}'.format(prev_ver))
+                if self.staged:
+                    click.echo("Collecting staged files only")
+                    all_changed_files_string = run_command('git diff --name-status --staged master')
+                else:
+                    # only changes against prev_ver (without local changes)
+                    all_changed_files_string = run_command('git diff --name-status {}'.format(prev_ver))
```

Comment:
If `staged` is set then it would return in the changes you made above - this code is unreachable.

Reply:
if staged is set, those the only things that should come back.